### DISCUSSION TITLE
Problems getting stdout and stderr to show up in debug console

### DISCUSSION THREAD

**Discussion by evdmeer** [devcontainers/community#140] [Q&A]
I am using dev containers inside Visual Studio code with the official microsoft devcontainers/php:1.2.3-8.3-bookworm container.
I am using the built-in apache2 install inside the container to develop an application. 
Apache inside the container is setup by default to log to /var/log/apache2/error.log , which is symlinked to /dev/stderr by default.

I have confirmed if I manually run the container via 'Docker run', that all apache/php warnings/errors do indeed get written to stderr and they do show up in my commandline if I run it that way.

I am expecting this output to appear in the debug console log inside vscode once something gets written to the apache error log (it should, according to what I have found online), but I can't get it to work.

I have experimented with removing the symlink from error.log to /dev/stderr and it writes to the log without problems if I do that.

Am I missing something?
Is there something I should configure to get this to work?
 
 My dockerfile:
```
# PHP 8.3
FROM mcr.microsoft.com/devcontainers/php:1.2.3-8.3-bookworm    

# Install MariaDB client
RUN apt-get update && export DEBIAN_FRONTEND=noninteractive \
    && apt-get install -y mariadb-client libicu-dev libpng-dev libxslt-dev libc-client-dev libkrb5-dev libmagickwand-dev libzip-dev sshpass \ 
    && apt-get clean -y && rm -rf /var/lib/apt/lists/*

# Install php extensions
RUN docker-php-ext-configure gd --with-freetype --with-jpeg=/usr/include/ --enable-gd
RUN docker-php-ext-install mysqli pdo pdo_mysql bcmath calendar exif ftp gettext gd sockets xsl intl zip soap
RUN pecl install imagick && docker-php-ext-enable imagick

# Custom PHP.ini for settings override
COPY ../docker-php-custom.ini /usr/local/etc/php/conf.d/

RUN a2enmod rewrite && a2enmod headers
```

docker-compose.yml
```
version: '3.8'

services: 
  app:
    user: vscode
    build:
      context: .
      dockerfile: Dockerfile

    volumes:
      - ../..:/workspaces:cached

    # Overrides default command so things don't shut down after the process ends.
    command: sleep infinity
```

devcontainer.json
```
// For format details, see https://aka.ms/devcontainer.json. For config options, see the
// README at: https://github.com/devcontainers/templates/tree/main/src/php-mariadb
{
	"name": "PHP & MariaDB",
	"dockerComposeFile": "docker-compose.yml",
	"service": "app",
	"workspaceFolder": "/workspaces/${localWorkspaceFolderBasename}",

	// Features to add to the dev container. More info: https://containers.dev/features.
	// "features": {},

	// For use with PHP or Apache (e.g.php -S localhost:8080 or apache2ctl start)
	"forwardPorts": [8080, 3306],

	"postStartCommand": "<snip> && apache2ctl start",
	
	"postAttachCommand": "<snip>",

}
```

**Comment by samruddhikhandale** [comment]
@gauravsaini04 / @prathameshzarkar9 Can either of you help with investigating this? Thanks!
